Sources/ChangeMenuBarColor/Commands/Abstract/Command.swift

```diff
@@ -56,6 +56,18 @@ class Command {
             Log.error("Cannot read the currently set macOS wallpaper. Try providing a specific wallpaper as a parameter instead.")
             return nil
         }
+        
+        var maxImageSize = NSSize(width: 0, height: 0)
+        for i in 0...(wallpaper.representations.count-1)
+        {
+                let rep = wallpaper.representations[i]
+                let imageSize = NSSize(width: rep.pixelsWide, height: rep.pixelsHigh)
+                if (imageSize.width >= maxImageSize.width) || (imageSize.height >= maxImageSize.height)
+                {
+                    maxImageSize = imageSize
+                }
+        }
+        wallpaper.size = maxImageSize
```

Comment:
I would probably write it as something like

```
wallpaper.size = wallpaper.representations.reduce(wallpaper.size) { size, representation in
    return CGSize(width: max(size.width, CGFloat(representation.pixelsWide)), height: max(size.height, CGFloat(representation.pixelsHigh)))
})
```

I just wonder why this needs to be done only here not also somewhere around the line 50 when loading the wallpaper from an explicitly specified file.

Reply:
Apart from having a closing bracket ")" too much at the end your code example works for me too.
So please choose what ever code you like the most.
I just prefer for-loops rather then using mapping, folding or reduce functions since I crew up with them (started with BASIC) and it makes it at least for me easier to quickly understand a code, thats all. 😅 And large-scale highly parallel data processing that usually justifies mapping functions is no factor here since at least in my example picture this array has only one element and I doubt there many images that uses more then maybe 3 or 5. 
This means I think there is no right or wrong how here, so please go ahead and change it.